### ISSUE TITLE
Bypass branch protection for automatic dependency updates

### ISSUE THREAD

**Issue by Oaphi** [userscripters/stackexchange-global-types#11]
Blocked by: https://github.com/community/community/discussions/13836

Branch protection is disabled until there is a non-hacky way of bypassing branch protections for GitHub Actions

**Comment by mjpieters** [comment]
You can have PRs auto-merge. See my [dependabot-auto-merge workflow](https://github.com/mjpieters/SO-userscripts/blob/main/.github/workflows/dependabot-auto-merge.yml) for the details, but it works like this:

- fetch dependabot PR metadata with [`dependabot/fetch-metadata`](https://github.com/marketplace/actions/fetch-metadata-from-dependabot-prs) and decide if you want to auto-merge the PR.
- use the `gh` command-line tool to approve the PR, and enable auto-merging.
- PRs are automatically merged into the main branch once tests pass.

It does require that you enable the *Allow auto-merge* setting in the repository.

My main branch is protected, it requires status checks to pass. My workflow auto-accepts patch and minor version updates for production and dev dependencies, and works absolutely bloody great! It also manages labels on the PRs so I can see at a glance what PRs need explicit approval.
